src/dmd/escape.d

```diff
@@ -1971,9 +1975,10 @@ void escapeByRef(Expression e, EscapeByResults* er, bool live = false)
                         {
                             Parameter p = tf.parameterList[i - j];
                             const stc = tf.parameterStorageClass(null, p);
-                            if ((stc & (STC.out_ | STC.ref_)) && (stc & STC.return_))
+                            ScopeRef psr = buildScopeRef(stc);
+                            if (psr == ScopeRef.ReturnRef || psr == ScopeRef.ReturnRef_Scope)
                                 arg.accept(this);
-                            else if ((stc & STC.scope_) && (stc & STC.return_))
+                            else if (psr == ScopeRef.ReturnScope || psr == ScopeRef.Ref_ReturnScope)
```

Comment:
How about separating lines 1978-1981 into a separate PR? There's too much going on in this PR making it hard to see the whole.

Reply:
I know this PR is big, which is why I made the separate FuncFlag PR already. I'll continue to split it up as you wish.

Reply:
https://github.com/dlang/dmd/pull/13802